src/runtime_src/core/tools/common/tests/TestHelper.cpp

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023-2024 Advanced Micro Devices, Inc. All rights reserved.
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "TestHelper.h"
+
+// Constructor for BO_set
+// BO_set is a collection of all the buffer objects so that the operations on all buffers can be done from a single object
+// Parameters:
+// - device: Reference to the xrt::device object
+// - kernel: Reference to the xrt::kernel object
+BO_set::BO_set(xrt::device& device, xrt::kernel& kernel, size_t buffer_size) 
+  : buffer_size(buffer_size) 
+{
+  // Initialize buffer objects with appropriate flags and group IDs
+  bo_instr = xrt::bo(device, buffer_size, XCL_BO_FLAGS_CACHEABLE, kernel.group_id(5));
+  bo_ifm = xrt::bo(device, buffer_size, XRT_BO_FLAGS_HOST_ONLY, kernel.group_id(1));
+  bo_param = xrt::bo(device, buffer_size, XRT_BO_FLAGS_HOST_ONLY, kernel.group_id(2));
+  bo_ofm = xrt::bo(device, buffer_size, XRT_BO_FLAGS_HOST_ONLY, kernel.group_id(3));
+  bo_inter = xrt::bo(device, buffer_size, XRT_BO_FLAGS_HOST_ONLY, kernel.group_id(4));
+  bo_mc = xrt::bo(device, buffer_size, XRT_BO_FLAGS_HOST_ONLY, kernel.group_id(7));
```

Comment:
Please use member initializer list to construct the bo objects

Reply:
That's a good idea ! 
I have updated the latest patch with this change.
Thanks